src/main/scala/viper/silver/ast/pretty/PrettyPrinter.scala

```diff
@@ -634,11 +631,20 @@ object FastPrettyPrinter extends FastPrettyPrinterBase with BracketPrettyPrinter
   }
 
   /** Show some node inside square braces (with nesting). */
-  def showBlock(stmt: Stmt): Cont = {
-    braces(nest(defaultIndent,
-      lineIfSomeNonEmpty(stmt match {case s: Seqn => s.scopedDecls; case _ => Seq()}, stmt.children) <>
-        showStmt(stmt)
-    ) <> line)
+  def showBlock(stmt: Stmt): Cont = stmt match {
+    case Seqn(stmts, scopedDecls) =>
+      val locals = scopedDecls.collect { case l: LocalVarDecl => l }
+      val nonEmptyStmts = stmts.filter {
+        case s@Seqn(ss, sd) => ss.nonEmpty || sd.nonEmpty || s.info.comment.nonEmpty
```

Comment:
I wonder if not filtering these empty statements would be better. For frontends, it may be useful sometimes to know whether we are generating too many empty stmts, which is not possible if we filter them. They could in principle be printed as "{ }"

Reply:
IIRC there was a specific reason for filtering them. Viper itself uses an empty ``Seqn`` as its representation for no-ops somewhere in the translator (to translate local var declarations?), so these would show up all over the place.